Задание 4 решать с помощью функции. 
Есть ли статистически значимые различия в росте дочерей?
Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169
Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160

In [5]:
import scipy.stats as stats
import numpy as np

In [6]:
x=np.array([172, 177, 158, 170, 178,175, 164, 160, 169]) # массив с ростом матерей
x

array([172, 177, 158, 170, 178, 175, 164, 160, 169])

In [7]:
y=np.array([173, 175, 162, 174, 175, 168, 155, 170, 160]) # массив с ростом взрослых дочерей
y

array([173, 175, 162, 174, 175, 168, 155, 170, 160])

По условию задания значение σ не известно.   
Гипотеза $\displaystyle H_0$: $µ = µ_0$   
Гипотеза $\displaystyle H_0$: $µ ≠ µ_0$   
Зададим α = 0,05    
Выбираем тест по критерию Стьюдента для независимых выборок.    
Воспользуемся функцией:

In [8]:
stats.ttest_ind(x,y)

Ttest_indResult(statistic=0.35581195161541973, pvalue=0.7266292679160495)

В результате получили $\displaystyle t_н$ = 0.3558 и p-value = 0.7266   
Вывод: т.к. p-value > α, принимаем гипотезу $\displaystyle H_0$